src/Cmfd.cpp

```diff
@@ -1750,22 +1774,278 @@ void Cmfd::setPolarQuadrature(PolarQuad* polar_quad) {
 
 
 /**
- * @brief Get the new to old flux ratio for a CMFD cell.
- * @param the CMFD cell ID
- * @param the MOC energy group
- * @return the flux ratio
+ * @brief Get the ratio of new to old CMFD cell flux in a given CMFD cell
+ *        and CMFD energy group containing a given MOC energy group.
+ * @param cmfd_cell The CMFD cell of interest.
+ * @param moc_group The MOC energy group of interest.
+ * @return The ratio of new to old CMFD cell flux.
  */
 FP_PRECISION Cmfd::getFluxRatio(int cmfd_cell, int moc_group) {
 
   int cmfd_group = _group_indices_map[moc_group];
   FP_PRECISION old_flux = _old_flux[cmfd_cell*_num_cmfd_groups + cmfd_group];
   FP_PRECISION new_flux = _new_flux[cmfd_cell*_num_cmfd_groups + cmfd_group];
-  FP_PRECISION ratio = new_flux / old_flux;
+  return new_flux / old_flux;
+}
+
+
+/**
+ * @brief Generate the k-nearest neighbor CMFD cell stencil for each FSR.
+ * @detail This method finds the k-nearest CMFD cell stencil for each FSR
+ *         and saves the stencil, ordered from the closest-to-furthest
+ *         CMFD cell, in the _k_nearest_stencils map. The stencil of cells
+ *         surrounding the current cell is defined as:
+ *
+ *                             6 7 8
+ *                             3 4 5
+ *                             0 1 2
+ *
+ *         where 4 is the given CMFD cell. If the cell is on the edge or corner
+ *         of the geometry and there are less than k nearest neighbor cells,
+ *         k is reduced to the number of neighbor cells for that instance.
+ */
+void Cmfd::generateKNearestStencils(){
+
+  std::vector< std::pair<int, FP_PRECISION> >::iterator iter2;
+  std::vector<int>::iterator iter;
+  Point* centroid;
+
+  /* Loop over mesh cells */
+  for (int i = 0; i < _num_x*_num_y; i++){
+
+    /* Loop over FRSs in mesh cell */
+    for (iter = _cell_fsrs.at(i).begin();
+         iter != _cell_fsrs.at(i).end(); ++iter) {
+
+      /* Get centroid */
+      centroid = _geometry->getFSRCentroid(*iter);
```

Comment:
It seems like ``*iter`` is used in quite a few places in this method. Could you add a local variable ``fsr_id = *iter`` and then use ``fsr_id`` instead of ``*iter`` to make this easier to comprehend at first sight?